Replace `PhpMyAdmin\Core::fatalError()` method calls

MauricioFauth opened the pull request:
The `PhpMyAdmin\Core::fatalError()` method terminates the execution with `exit` when called.
This PR replaces the calls to `fatalError` with other options, like throwing an exception for example.

MauricioFauth committed:
Remove Core::fatalError call from SchemaExportController

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

MauricioFauth committed:
Replace Core::fatalError with an exception in Export::processExportSchema()

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

MauricioFauth committed:
Replace Core::fatalError() calls from the Common class

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

williamdes commented on libraries/classes/Common.php:
```diff
@@ -225,13 +239,17 @@ public static function run(bool $isSetupPage = false): void
+                echo (new Template())->render('error/generic', [
+                    'lang' => $GLOBALS['lang'] ?? 'en',
+                    'dir' => $GLOBALS['text_dir'] ?? 'ltr',
+                    'error_message' => sprintf(
+                        __('You should upgrade to %s %s or later.'),
                         'MySQL',
-                        $GLOBALS['cfg']['MysqlMinVersion']['human'],
-                    ]
-                );
+                        (string) $GLOBALS['cfg']['MysqlMinVersion']['human']
+                    ),
+                ]);
```
the render pattern is duplicated, maybe a method to render an error would be better ?

MauricioFauth replied:
That's true. However I'll handle it later when introducing the PSR-7 Response class. Something like an ErrorResponse class.

williamdes commented on libraries/classes/Common.php:
```diff
@@ -535,11 +551,9 @@ private static function checkServerConfiguration(): void
-                'The ini_get and/or ini_set functions are disabled in php.ini. phpMyAdmin requires these functions!'
-            )
-        );
+        throw new Exception(__(
```
could we have an internal exception class to avoid catching the entire world afterwards ?

MauricioFauth committed:
Replace fatalError() with exception in Core::warnMissingExtension()

Also moves some other calls to Core::warnMissingExtension() to the
Common::checkRequiredPhpExtensions() method.

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

MauricioFauth committed:
Replace Core::fatalError() calls in controllers

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

MauricioFauth committed:
Remove $message_args param from Core::fatalError()

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

MauricioFauth committed:
Remove Core::fatalError method

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

MauricioFauth committed:
Use more specific exceptions for some errors

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>
